lightning/src/ln/channelmanager.rs

```diff
@@ -2146,11 +2156,16 @@ impl<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> ChannelMana
 							msg: update
 						});
 					}
+					did_work = true;
 					chan.set_update_status(UpdateStatus::Enabled);
 				},
 				_ => {},
 			}
 		}
+
+		if !did_work {
+			persistence_guard.persistence_update = false;
+		}
```

Comment:
`PersistenceNotifierGuard` is meant to ensure that persisters are notified. Rather than adding complexity to the guard and obfuscating the usage, it seems more appropriate to use `total_consistency_lock` directly and conditionally call `persistence_notifier` if needed.

Reply:
Hmm, I kind of like the explicit-ness of having to say "no, I know I'm supposed to notify here, but I actually don't want to" vs putting it at the top of the method without it being explicit. Would you prefer if we hide the bool inside a method like `unlock_no_notify`?

Reply:
To me, the act of using `PersistenceNotifierGuard` signals to the reader that notification will occur. So it would be unexpected if somewhere later the code were to disables that. Especially if the common case is to _not_ notify as it seems like is the case in this method (i.e., only notify when the channel's liveness changes).

Reply:
Right, I don't disagree, but I do think it would be bad to start taking the `total_consistency_lock` directly again as that'd be an easy thing to miss because its not obvious that you maybe do want to notify when reading that code. I guess my question is more if there's a refactor/rename of `PersistenceNotifierGuard` that makes more sense.

Reply:
Yeah, so the (read) guard is really for `total_consistency_lock` and as a side effect notifies persisters. So maybe a rename with two constructors to make it explicit like `TotalConsistencyGuard::notify_by default(..., ...)` and `TotalConsistencyGuard::notify_conditionally(...)` with a `notify_on_unlock` method take the notifier by reference when you want to explicitly enable persistence. Or something akin to that -- would be nice if "persistence" were spelled out without being too verbose.

Internally, the notifier could be an `Option` to prevent accidentally persisting.

Reply:
Or maybe `always_notify` and `conditionally_notify` if there is only a way to enable (and not disable) after construction.